precompiles/avs/query.go

```
@@ -102,7 +115,111 @@ func (p Precompile) GetOperatorOptedUSDValue(
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, "string", operatorAddr)
+	}
+
+	param, err := sdk.AccAddressFromBech32(operatorAddr)
+	if err != nil {
+		return nil, err
+	}
+	flag := p.avsKeeper.GetOperatorKeeper().IsOperator(ctx, param)
+
+	return method.Outputs.Pack(flag)
+}
+
+func (p Precompile) GetTaskInfo(
+	ctx sdk.Context,
+	_ *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != len(p.ABI.Methods[MethodGetTaskInfo].Inputs) {
+		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, len(p.ABI.Methods[MethodGetTaskInfo].Inputs), len(args))
+	}
+	addr, ok := args[0].(common.Address)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, "common.Address", addr)
+	}
+	taskID, ok := args[1].(uint64)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 1, "uint64", taskID)
+	}
+
+	task, err := p.avsKeeper.GetTaskInfo(ctx, strconv.FormatUint(taskID, 10), addr.String())
+	if err != nil {
+		// if the avs does not exist, return empty array
+		if errors.Is(err, avstype.ErrNoKeyInTheStore) {
+			return method.Outputs.Pack("")
+		}
+		return nil, err
+	}
+	info := []uint64{task.StartingEpoch, task.TaskResponsePeriod, task.TaskStatisticalPeriod}
+
+	return method.Outputs.Pack(info)
+}
+
+// GetCurrentEpoch obtain the specified current epoch based on epochIdentifier.
+func (p Precompile) GetCurrentEpoch(
+	ctx sdk.Context,
+	_ *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != len(p.ABI.Methods[MethodGetCurrentEpoch].Inputs) {
+		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, len(p.ABI.Methods[MethodGetCurrentEpoch].Inputs), len(args))
+	}
+	epochIdentifier, ok := args[0].(string)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, "string", epochIdentifier)
+	}
+	epoch, flag := p.avsKeeper.GetEpochKeeper().GetEpochInfo(ctx, epochIdentifier)
+	if !flag {
+		return nil, nil
```

Should return an error here: `nil, err`